monkey/infection_monkey/telemetry/messengers/batching_telemetry_messenger.py

```diff
@@ -0,0 +1,94 @@
+import queue
+import threading
+import time
+from typing import Dict
+
+from infection_monkey.telemetry.i_batchable_telem import IBatchableTelem
+from infection_monkey.telemetry.i_telem import ITelem
+from infection_monkey.telemetry.messengers.i_telemetry_messenger import ITelemetryMessenger
+
+DEFAULT_PERIOD = 5
+WAKES_PER_PERIOD = 4
+
+
+class BatchingTelemetryMessenger(ITelemetryMessenger):
```

Comment:
What's the purpose of implementing the ITelemetryMessenger interface? We already receive an `ITelemetryMessenger` object, so this class will know it's interface. Can't we do something like `telemetry_messenger.send_telemetry = self.add_telemetry_to_queue` in the constructor? This would also solve the issue that `send_telemetry` method is not sending anything, in fact, it's just adding the telemetry to a queue

Reply:
>  What's the purpose of implementing the ITelemetryMessenger interface?
In implementing ITelemetryMessenger, we're stating explicitly that this object must conform to that interface.

> We already receive an ITelemetryMessenger object, so this class will know it's interface.
Just because this class "has-a" ITelemetryMessenger doesn't mean it implements ITelemetryMessenger.

> Can't we do something like telemetry_messenger.send_telemetry = self.add_telemetry_to_queue in the constructor?
I think this would make the code harder to read. It implements ITelemetryMessenger, but where is it's `send_telemetry()` method? I have to read the constructor to find out _how_ it implements ITelemetryMessenger.

> This would also solve the issue that send_telemetry method is not sending anything, in fact, it's just adding the telemetry to a queue.
I'm not sure I see what problem needs to be solved. For anyone who uses the class, all they care about is that telemetry is sent. The purpose of ITelemetryMessenger is that they don't need to care how it is sent.